modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -2873,8 +2868,22 @@ record = new TxRecord(tx.state(), tx.nearXidVersion(), tx.writeVersion(), nodes)
         }
     }
 
-    /** Checks if a record belongs to a persistent cache with WAL enabled. */
-    private boolean containsCacheWithEnabledWal(IgniteTxAdapter tx) {
+    /**
+     * Returns {@code true} if it is possible to skip logging of transaction record to WAL.
+     *
```

Comment:
I would expect to see the criteria in javadoc: why tx logging can be omitted?

Reply:
Please check the updated javadoc:
```
    /**
     * Returns {@code true} if the given transaction should be logged in WAL and {@code false} otherwise.
     * Transaction marker should be logged in WAL if the following conditions are met:
     * <ul>
     *     <li>local node is a part of baseline topology.</li>
     *     <li>point-in-time recovery is enabled,
     *     or there is a cache enlisted in the transaction with persistence and WAL enabled for the corresponding cache group.</li>
     * </ul>
     * @param tx Transaction.
     * @return {@code true} if the given transaction should be logged in WAL.
     */
    private boolean txRecordLoggingNeeded(IgniteTxAdapter tx) {
```